mlx/backend/common/primitives.cpp

```diff
@@ -94,6 +94,19 @@ void ArcTan::eval(const std::vector<array>& inputs, array& out) {
   }
 }
 
+void ArcTan2::eval(const std::vector<array>& inputs, array& out) {
+  assert(inputs.size() == 2);
+  const auto& in1 = inputs[0];
+  const auto& in2 = inputs[1];
+  if (issubdtype(out.dtype(), inexact)) {
+    binary(in1, in2, out, detail::ArcTan2());
+  } else {
+    throw std::invalid_argument(
+        "[arctan2] Cannot compute inverse tangent of elements in array"
+        " with non floating point type.");
+  }
+}
```

Comment:
Move this to `binary.cpp` and follow the example of `LogAddExp` to only compile for floating point types: https://github.com/ml-explore/mlx/blob/main/mlx/backend/common/binary.cpp#L178-L197